### ISSUE TITLE
look into bleeding rate and pain in rewrite

### ISSUE THREAD

**Issue by PabstMirror** [acemod/ACE3#6906]

**Comment by alganthe** [comment]
A large avulsion or velocity wound to the leg results in death after exactly 40s, it's extreme and results in an insane casualty rate in actual gameplay, it definitely needs to be toned down.

**Comment by SilentSpike** [comment]
Commented on #6941 for how I think we can definitively solve the bloodloss issues.

**Comment by SilentSpike** [comment]
As I commented on PR:

>I think in order to correctly solve this bleeding issue, we first need to choose time periods that are reasonable to survive one of each given wound type and then set the bleeding values and modifier calculations so that they average those time scales.

Following the bleeding calculations through, the critical path to death is:

1. Wound bleeding is updated upon wounding/bandaging/etc. This is where bleeding is scaled per body part and a maximum value of 1 is possible (so it's a percentage). See `medical_status_fnc_updateWoundBloodLoss`.
2. Bloodloss is calculated as a product of wound bleeding, cardiac output and the bleeding coefficient setting (this must be litres/second for following calculations to work). If heart rate is normal, cardiac output has a maximum value of 5.25 when blood volume is default 6 litres and this decreases with blood volume. See `medical_status_fnc_getBloodLoss` and `medical_status_fnc_getCardiacOutput`.
3. Unit vitals loop periodically calculates the change in blood volume as a product of time between loops and bloodloss value (hence should be litres). This is assuming the absence of bloodbags, they're factored in here. See `medical_status_fnc_getBloodVolumeChange`.
4. If blood volume is less than 50% unit will die via bleeding out.

With this, we can take the value of a given wound type and follow through the calculations (assuming default heart rate, no bloodbags, and wound is applied to torso) to figure out how long it would take to cause bleedout.

**Comment by SilentSpike** [comment]
Based on the calculations, I can produce the following plot which shows the average time it would take to bleedout for each wound type when applied to the torso (assuming wound is fully open, heart rate is normal throughout, bleeding coefficient setting is 1, time between vital loop iterations is 1s, nastiness modifier is taken as worst case average of `0.25 + 8 * exp(-5/0.5)`).

![image](https://user-images.githubusercontent.com/5459452/57969396-5a4efc00-796e-11e9-964d-54cee9b32f3b.png)

When adjusted for the values in PR #6941 these become

![image](https://user-images.githubusercontent.com/5459452/57969501-7dc67680-796f-11e9-8a47-8e140d738bb9.png)

**Comment by SilentSpike** [comment]
So I followed all the bleeding maths through (see above) and by making some assumptions along the way predictions can be made for how long a unit will survive with a given wound type.

```matlab
% Assuming a fully open wound
wound_types = {'Velocity', 'Puncture/Laceration/Crush', 'Cut', 'Avulsion'};
bleeding_values = [0.5, 0.05, 0.04, 0.25]; % unknown units

% Arms limit at 0.3, legs at 0.5, head at 0.9 and torso at 1
apply_to_part = 1;
wound_bleeding_lims = [1, 0.9, 0.5, 0.3];

% Applying worst case nastiness modifier on average
nastiness = 0.25 + 8 * exp(-5/0.5);

figure
hold on
for bleeding = bleeding_values
    blood_volume = 6; % litre
    deltaT = 1; % s
    
    wound_bleeding = bleeding * nastiness;

    % There's an overall cap at 1
    wound_bleeding = min(wound_bleeding, ...
        wound_bleeding_lims(apply_to_part));

    % Death occurs at 50% blood volume
    while (blood_volume(end) > 3)
        % Heartrate is assumed to be normal, so cardiac output is a function of
        % blood volume ratio only
        cardiac_output = (blood_volume(end)/6) / 0.1904761;

        bloodloss = cardiac_output * wound_bleeding;

        blood_volume(end+1) = blood_volume(end) - bloodloss * deltaT;
    end
    
    % Add it to the plot
    plot((0:length(blood_volume)-1)*deltaT,blood_volume)
end

title('Average Bleedout Time for Torso Wounds')
xlabel('Time (seconds)')
ylabel('Blood Volume (litres)')
legend(wound_types)
```

I implemented this as the above MATLAB code which produces plots such as

`deltaT = 1`
![image](https://user-images.githubusercontent.com/5459452/57985377-cdd03680-7a5e-11e9-8b43-784ad772003a.png)

`deltaT = 3`
![image](https://user-images.githubusercontent.com/5459452/57985420-bb0a3180-7a5f-11e9-983b-bd8f57c92758.png)

`deltaT = 10`
![image](https://user-images.githubusercontent.com/5459452/57985423-d117f200-7a5f-11e9-8abb-eb7ef8013086.png)

This is good news, it suggests if we know how long a unit should survive, we can tweak the wound values to adjust this overall time as desired. 

Caveats:
- Important to note that I've ignored heart rate assuming it stays normal which makes the cardiac output calculation easier. However, the heart rate is also a function of the current blood volume level so I expect in-game results to be somewhat different from what these graphs suggest.
- Units will go into cardiac arrest at 60% blood volume, so bleeding will actually cease at this point and not continue straight through to death as it looks here (unless the just goes from >60% to <50% which I believe could actually result in instant death).
- `deltaT` varies in the range 1-10 depending on the number of units and other factors that slow down the vitals loop, so the actual result will be some superposition of all these graphs. There isn't too much variance though.
- The above graphs are for an isolated wound on the torso which is fully open and has the worst nastiness on average. I think before we tweak the wound config values we need to revisit the nastiness calculations as Pabst mentioned https://github.com/acemod/ACE3/pull/6941#issuecomment-491539879. I'm looking at it now and can't quite see how the `linearConversion` from 0-20 makes sense because it will almost always be 0.5.

**Comment by SilentSpike** [comment]
Now messing around with making changes in-game and this won't be as simple as I hoped.

Thinking about the calculations more I'm seeing more and more why it's currently so punishing:
- Cardiac output is blood pumped per second in litres
- Multiplied by wound bleeding gives blood lost per second
- Therefore wound bleeding is effectively what percentage of your cardiac output is being bled
- So a single shot to the arm can result in 30% blood loss and a single shot to the torso can result in 50% blood loss (with current config values)

The only thing scaling that down is the bleeding modifier in the nastiness stuff (nastiness basically determines size of the wound).

Also worth noting I forgot damage can be greater than 1, which means the graphs above aren't using the worst case nastiness, but actually the **best** case nastiness (however, typically it will be on the lower end of the scale which is where the best case is).
